### PR TITLE
GH-10932 Migrate 'Session.UpdateLastActivityAt' to Sync by default

### PR DESCRIPTION
**Pull request by lurcio**
#### Summary
Migrates "Session.UpdateLastActivityAt" to Sync by default

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/10932

### CODE REVIEW

File: store/storetest/session_store.go [hunk 1]
```diff
@@ -258,9 +258,8 @@ func testSessionStoreUpdateLastActivityAt(t *testing.T, ss store.Store) {
 	s1, err := ss.Session().Save(s1)
 	require.Nil(t, err)
 
-	if err := (<-ss.Session().UpdateLastActivityAt(s1.Id, 1234567890)).Err; err != nil {
-		t.Fatal(err)
-	}
+	_, err = ss.Session().UpdateLastActivityAt(s1.Id, 1234567890)
```
Comment (lurcio) [inline store/storetest/session_store.go]: I'm not sure whether the return value should be checked here

**Comment by lurcio** [comment]
I'll look at the build failure later today

**Comment by jespino** [comment]
@lurcio you need to migrate the `app/session.go` and `app/session_test.go` usages of `UpdateLastActivityAt`, and while I was checking that, I realized that the returned sessionId is not used anywhere, probably makes sense to return just a `*model.AppError` instead of `(string, *model.AppError)`. But because this PR is about "migrating existing behavior to sync by default" is not necessarily in the scope of this PR, so, feel free to decide to add the change or not to this PR.

**Comment by lurcio** [comment]
@jespino thank you for the advice, I think I've fixed the issues now.

**Comment by jespino** [comment]
Merged! Thanks @lurcio. :tada:

**Pull request merged**